pkg/runtime/k0s/delete_masters.go

```diff
@@ -18,113 +18,50 @@ import (
 	"context"
 	"fmt"
 	"net"
-	"strings"
 
-	"github.com/sealerio/sealer/common"
-	"github.com/sealerio/sealer/pkg/client/k8s"
-
-	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	"golang.org/x/sync/errgroup"
 )
 
-func (k *Runtime) deleteMasters(masters []net.IP) error {
-	if len(masters) == 0 {
-		return nil
+func (k *Runtime) deleteMasters(mastersToDelete, remainMasters []net.IP) error {
+	var remainMaster0 *net.IP
+	if len(remainMasters) > 0 {
+		remainMaster0 = &remainMasters[0]
+		logrus.Infof("Master changed, remain master0 is: %s", remainMaster0)
 	}
+
 	eg, _ := errgroup.WithContext(context.Background())
-	for _, master := range masters {
-		master := master
+	for _, m := range mastersToDelete {
+		master := m
```

Comment:
no need to deep copy a variable here, just as the params for go func()

Reply:
I'm afraid I don't totally agree. we need a need copy variable, if not, `master` always is the first one.

Reply:
@kakaZhou719 

FYI:

https://david-yappeter.medium.com/golang-pass-by-value-vs-pass-by-reference-e48aac8b2716#:~:text=Pass%20by%20value%20will%20pass,affect%20the%20original%20variable%20value.

https://go.dev/play/p/28vulG8TOeb